tests/README.md

```diff
@@ -1,14 +1,64 @@
-Functional Tests for Runtime Node
-=================================
+# Functional Tests for Runtime Node
 
-## Run unit tests
-1. setup a relaychain/parachain deployment locally, for example, using `polkadot-launch`. 
-2. `yarn install`
-3. `yarn correctness_test`
-You can pass an optional argument `--address=<some_address>` but the default is set to  `ws://127.0.0.1:9800`.
+## Prerequisites
+
+- nodejs >= 16.x
+- yarn
+
+## MantaPay
+
+### correctness test for rpc methods
+
+1. setup a relaychain/parachain deployment locally, for example, using `polkadot-launch`.
+2. Navigate to [config.json](./config/config.json), ensure `nodeAddress` points to the parachain.
+3. Build tests
+
+```
+cd tests
+yarn
+```
+
+4. Register new assets for future testing.
+
+```
+yarn init-assets
+```
+
+This script will register new assets, and issue some tokens to each assets.
+
+5. Submit some private transactions to generate some Utxos.
+
+```
+yarn mint-assets
+```
 
-## Run rpc performance tests
-1. setup a relaychain/parachain deployment locally, for example, using `polkadot-launch`. 
+6. Now validate two rpc methods `pull_ledger_diff` and `dense_pull_ledger_diff`, ensure their response is the same.
+
+```
+yarn validate-pull-ledger-diff
+```
+
+Ensure test case get passed.
+
+### performance tests rpc methods
+
+1. setup a relaychain/parachain deployment locally, for example, using `polkadot-launch`.
 2. `yarn install`
 3. `yarn performance_test`
-You can pass an optional argument `--address=<some_address>` but the default is set to  `ws://127.0.0.1:9800`.
+   You can pass an optional argument `--address=<some_address>` but the default is set to `ws://127.0.0.1:9800`.
+
+### private transaction tests
+
+todo!()
+
+## Common Tests
+
+todo!()
```

Comment:
We have runtime upgrade test if that counts as  common, common may be a bit ambiguous

Reply:
Agreed.

Reply:
Renamed the `common` folder as `runtime`.

Reply:
update README too?

Reply:
Done